### PR TITLE
Adding pushbullet to the list of notification providers

### PR DESCRIPTION
**Pull request by davidrothera**
Added https://www.pushbullet.com to the list of supported Providers.

Checked and tested with all methods (test, snatch, downloaded)

### CODE REVIEW

**Comment by davidrothera** [comment]
This is as per requested in https://code.google.com/p/sickbeard/issues/detail?id=2441

**Comment by thezoggy** [comment]
just to note
https://github.com/mr-orange/Sick-Beard/commit/1b3fe26625e630f6adeddfc436cf98de183a3ad4

you really should squash and rebase your commits

**Comment by davidrothera** [comment]
Squashed commits as requested

**Comment by rtjust** [comment]
Hah I just did the same thing this week. I started just before you posted this though :) Was a fun exercise at least. Looking forward to it.

**Comment by cnanney** [comment]
Looking forward to this being merged.

**Comment by Thiels851** [comment]
Yes, please merge this!

**Comment by davidrothera** [comment]
Updated to the latest cut of the `development` branch and then made adjustments to use v2 of the API endpoint as v1 is no longer supported.

**Comment by Stanzilla** [comment]
+1

**Comment by davidrothera** [comment]
Also squashed this to a single commit to make it easier to rebase

**Comment by MrAdam** [comment]
Any news on this? - been waiting on this for a while.

File: sickbeard/__init__.py [hunk 1]
```diff
@@ -355,7 +360,8 @@ def initialize(consoleLogging=True):
                 USE_LISTVIEW, METADATA_XBMC, METADATA_XBMC_12PLUS, METADATA_MEDIABROWSER, METADATA_MEDE8ER, METADATA_PS3, metadata_provider_dict, \
                 GIT_PATH, MOVE_ASSOCIATED_FILES, FILTER_ASSOCIATED_FILES, \
                 COMING_EPS_LAYOUT, COMING_EPS_SORT, COMING_EPS_DISPLAY_PAUSED, METADATA_WDTV, METADATA_TIVO, IGNORE_WORDS, CREATE_MISSING_SHOW_DIRS, \
-                ADD_SHOWS_WO_DIR, ANON_REDIRECT
+                ADD_SHOWS_WO_DIR, ANON_REDIRECT, USE_PUSHBULLET, PUSHBULLET_NOTIFY_ONSNATCH, PUSHBULLET_NOTIFY_ONDOWNLOAD, PUSHBULLET_APIKEY
+
```
Comment (ghost) [inline sickbeard/__init__.py:364]: I think this might be an unnecessary line break

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -1904,6 +1910,16 @@ def testPushover(self, userKey=None):
             return "Error sending Pushover notification"
 
     @cherrypy.expose
```
Comment (ghost) [inline sickbeard/webserve.py:1912]: I think that this is part of the next clause, so it should be after the edit.

**Comment by ghost** [comment]
I don't really care about this, but a device field might be nice.

**Comment by ghost** [comment]
Also, why not use Oauth for the user key? 
https://docs.pushbullet.com/oauth2

**Comment by davidrothera** [comment]
@thezoggy - I was following the same layout as other notifiers and using one file under the `sickbeard/notifiers` folder for the Python logic for the actual provider, I'm not using a separate lib, I know there was another diff like this that did however.

@erasmuswill - The part with `@cherrypy.expose` was done like that automatically by git, I know that it should be part of the next clause and the one above my code as part of my clause, this this still has the same outcome though.

As for why I haven't implemented Oauth or the device field, I think it would be better to ship a version that people can use and then iterate with extra features later down the line?

**Comment by chinedu40** [comment]
This code works perfectly after manually merging it. The only thing you could add, like couchpotato did, is a link to find out where to get your pushbullet access token, but this is unnecessary and can be added later. From a noobs perspective this code is ready to be merged.

**Comment by MrAdam** [comment]
Tried merging manually, works perfectly - when can we see this merged?

**Comment by SamJongenelen** [comment]
Same here, works like a charm

**Comment by cthulu** [comment]
Pls lets get this merged

**Comment by ronaldscott** [comment]
Another vote to merge this. Works great.

**Pull request closed without merging**